### PR TITLE
updated form description field. added custom logic and styling

### PR DESCRIPTION
**Pull request by kildre**
Updated the Description field on form to only allow 200 characters, but allow user to paste/enter more characters.
## Reference Issues or PRs
https://gitlab.jatic.net/jatic/team-metrostar/t-e-platform/-/issues/523

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [ ] Did you test the pull request locally?
- [ ] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

## Any other comments?

DESCRIPTION

### CODE REVIEW

**Comment by aktech** [comment]
@kildre can you merge main and see if that fixes CI issues? I just merged a PR for the same.

**Comment by kildre** [comment]
@aktech I'll do that.

**Comment by aktech** [comment]
The test failures are genuine failures. The JupyterHub doesn't loads:

Screenshot 2024-05-06 at 3 30 05 pm

Console errors (Its unable to find static files, as you have moved them):

Screenshot 2024-05-06 at 3 29 49 pm

File: ui/src/components/app-form/app-form.tsx [hunk 1]
```diff
@@ -124,8 +138,8 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
     formState: { errors },
   } = useForm<AppFormInput>({
     defaultValues: {
-      display_name: '',
-      description: '',
+      display_name: formData?.user_options.display_name || '',
+      description: formData?.user_options.description || '',
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:142]: These shouldn't be set here, they should be set in the useEffect here: https://github.com/nebari-dev/jhub-apps/blob/adfd708310c970afbe935b49b185eb2fa5174bac/ui/src/components/app-form/app-form.tsx#L342

File: ui/src/components/app-form/app-form.tsx [hunk 2]
```diff
@@ -139,6 +153,51 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
...
+  const descriptionRef = useRef(null); // Create a ref for the textarea
+
+  useEffect(() => {
+    const input = descriptionRef.current as HTMLInputElement | null;
+    if (input) {
+      const value = input.value;
+      const exceedLength = 200;
+
+      if (value.length > exceedLength) {
+        console.log('value', value.length, '/', exceedLength);
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:187]: Remove this

File: ui/src/components/app-form/app-form.tsx [hunk 3]
```diff
@@ -282,6 +341,8 @@ export const AppForm = ({ id }: AppFormProps): React.ReactElement => {
   // Populate form with existing app data
   useEffect(() => {
     if (formData?.name && formData?.user_options) {
+      console.log('formData', formData);
```
Comment (jbouder) [inline ui/src/components/app-form/app-form.tsx:344]: Remove this

**Review by jbouder — CHANGES_REQUESTED** [review]
Great work! Need to update a few things.

File: ui/src/components/app-form/app-form.tsx (hunk 3 above)
Comment (kildre) [reply ui/src/components/app-form/app-form.tsx]: Removed

**Review by kildre — COMMENTED** [review]
Made updates

**Review by jbouder — APPROVED** [review]
💯

**Pull request merged**